core/dictgen/src/SelectionRules.cxx

```diff
@@ -741,8 +738,7 @@ const ClassSelectionRule *SelectionRules::IsClassSelected(const clang::Decl* D,
 
             if (!isLinkDefFile) {
                // in genreflex - we could explicitly select classes from other source files
-               if (match == BaseSelectionRule::kFile) ++fFileNo; // if we have veto because of class defined in other source file -> implicit No
-               else {
+               if (match != BaseSelectionRule::kFile) {
```

Comment:
```suggestion
               // if we have veto because of class defined in other source file -> implicit No
               if (match != BaseSelectionRule::kFile) {
```